File: src/main/java/com/jack/Hello.java
package com.jack;

public class Hello {
    public static void main(String[] args) {
        System.out.println("Hello Java !!!");
        Person p = new Person(70.1f, 1.81f);
        Person p2 = new Person("Jack", 75.1f, 1.81f);
        p.hello();
        System.out.println(p.bmi());
        System.out.println(p2.bmi());
        // 基本資料型態 primitive types
        /*short s = 0;                    // 16 bits
        int i = 1;                      // 32 bits
        long l = 2;                     // 64 bits
        char c = 'j';                   // 16 bits
        byte b = 120;                   //  8 bits
        float f = 70.1f;                // 32 bits
        double d = 180.1;               // 64 bits
        boolean adult = true;
        // 參照資料型態 Reference data type
        String name = "Jack";
        Integer age = 27;
        Character c2 = 'a';
        Float f2 = 81.5f;
        Double d2 = 12.2;*/
    }
}

File: src/main/kotlin/com/jack/Hello.kt
package com.jack

fun main () {
    var s: String? = "abcde"
    println(s?.length)       // 安全呼叫，結果為 5
    println(s?.get(3))       // 安全呼叫，結果為 'd'
    println(s?.substring(3)) // 安全呼叫，結果為 "de"

    s = null
    println(s?.length)       // 結果為 null
    // println(s!!.length)   // 會拋出 NullPointerException

//    println("Hello Kotlin !!!");
//    Human().hello();

    val h = Human(weight = 7.5f, height = 1.81f) // 建構一個 human 物件
    println(h.bmi())
    h.hello()
    val score = 88
    println(score > 60)
    val c : Char = 'A'
    print(c.code > 60)
    print(c.toInt() > 60)
    /*val 名稱 : 型態 = 值 ( age / age2 的意思是一樣的 )
    val age : Int = 27;
    val age2 = 28;
    val name = "Jack";
    var nam2 : String;
    nam2 = "Hank";*/
}
// primary constructor
// 設計成需要三個變數才能建構所以 val h 會報錯
//如果 name 要放在最前面，那麼可以向上面一樣宣告身高體中的值，如果寫在最後面則不用
class Human(var name: String = "", var weight: Float, var height: Float) {
    init {
        println("test $weight")
    }
    // secondary constructor
//    constructor(name: String, weight: Float, height: Float) : this(weight, height)
    fun bmi() : Float {
        var bmi = weight / (height * height)
        return  bmi
    }

    fun hello () {
        println("Hello Human !!!")
    }
}